pkg/querier/worker/worker.go

```diff
@@ -135,8 +129,8 @@ func NewQuerierWorker(cfg Config, handler RequestHandler, log log.Logger, reg pr
 	case cfg.FrontendAddress != "":
 		level.Info(log).Log("msg", "Starting querier worker connected to query-frontend", "frontend", cfg.FrontendAddress)
 
-		factory = func(receiver serviceDiscoveryNotifications) (services.Service, error) {
-			return util.NewDNSWatcher(cfg.FrontendAddress, cfg.DNSLookupPeriod, receiver)
+		factory = func(receiver servicediscovery.Notifications) (services.Service, error) {
+			return servicediscovery.NewDNSServiceDiscovery(cfg.FrontendAddress, cfg.DNSLookupPeriod, receiver)
```

Comment:
I think `ServiceDiscovery` is redundant in the `servicediscovery` package. Commenting here as here's where it's obvious.

Reply:
WDYT about https://github.com/grafana/mimir/pull/2991/commits/eadd2987ebdb8dd98563df018c611b12891d9457?

Reply:
Love it!